### PR TITLE
[Crowdstrike] Adding many new event types

### PR DESCRIPTION
**Pull request by P1llus**
## What does this PR do?

This PR does some minor nits on the existing ingest pipelines, and adds support for new events.

Added support for:
- CSPM Streaming events
- CSPM Search events
- IDP Incidents
- IDP Summary events
- Mobile Detection events
- Recon Notification events
- XDR Detection events

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates [3358](https://github.com/elastic/sdh-beats/issues/3358)

### CODE REVIEW

File: packages/crowdstrike/data_stream/falcon/_dev/test/pipeline/test-falcon-detection-summary.log-expected.json [hunk 1]
```diff
@@ -60,7 +61,6 @@
                         "SuspendProcess": false
                     },
                     "PatternDispositionValue": 2304,
-                    "PatternId": "5728",
```
Comment (efd6) [inline packages/crowdstrike/data_stream/falcon/_dev/test/pipeline/test-falcon-detection-summary.log-expected.json:63]: Why is this removed? I don't see it reflected in any ECS field.

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/default.yml [hunk 2]
```diff
@@ -237,25 +276,55 @@ processors:
       if: "ctx.tags == null || !(ctx.tags.contains('preserve_original_event'))"
       ignore_failure: true
       ignore_missing: true
+  # This removes any fields that are mapped to ECS, but have not been renamed.
+  # This is to prevent the fields from being duplicated in the event.
```
Comment (efd6) [inline packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/default.yml:280]: The following fields added to this remove processor are not copied anywhere AFAICS (or not in all paths, e.g. `PatternID`):

- `crowdstrike.event.PatternId`
- `crowdstrike.event.TacticIds`
- `crowdstrike.event.AttemptOutcome`
- `crowdstrike.event.EventCreatedTimestamp`
- `crowdstrike.event.Disposition`
- `crowdstrike.event.MatchedTimestamp`

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/identity_protection_incident.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      type: string
+      tag: convert_start_time_epoch
+      ignore_missing: true
+      if: ctx.crowdstrike?.event?.StartTime != null
+  - gsub:
+      field: crowdstrike.event.StartTime
+      pattern: "\\d{6}$"
+      replacement: ""
+      tag: gsub_start_time_epoch
+      if: "ctx.crowdstrike?.event?.StartTime != null && ctx.crowdstrike.event.StartTime.length() > 18"
```
Comment (efd6) [inline packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/identity_protection_incident.yml:66]: It feels to me like arithmetic with REs is likely to be less efficient than with integers.

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/cspm_events.yml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+      tag: append_technique_name
+      if: ctx.crowdstrike?.event?.Technique != null
+on_failure:
+  - append:
+      field: error.message
+      value: |-
+        Processor "{{ _ingest.on_failure_processor_type }}" with tag "{{ _ingest.on_failure_processor_tag }}" in pipeline "{{ _ingest.on_failure_pipeline }}" failed with message "{{ _ingest.on_failure_message }}"
+  - set:
+      field: event.kind
+      value: pipeline_error
```
Comment (efd6) [inline packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/cspm_events.yml:163]: nit: final newlines?

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  - rename:
+      field: crowdstrike.event.Description
+      target_field: message
+      ignore_missing: true
+      tag: rename_description
+  - split:
+      field: crowdstrike.event.DataDomains
+      separator: ","
+      tag: split_data_domains
+      if: ctx.crowdstrike?.event?.DataDomains != null && ctx.crowdstrike?.event?.DataDomains.contains(",")
```
Comment (efd6) [inline packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:52]: Are the `contains` checks here necessary? Is it to avoid the cost of RE construction?

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/default.yml (hunk 2 above)
Comment (P1llus) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/default.yml:280]: PatternId seems like a mistake, a bit unsure why unless it was a duplicate of another field, I have changed this.
TacticIds are correct, however the name in the xdr summary pipeline was wrong.
AttemptedOutcome is used for the `event.outcome` in the IPD pipeline.
EventCreatedTimestamp is used in the CPSM pipeline.
Disposition is used for event.outcome in the CPSM pipeline.
MatchedTimestamp Is used in the Recon pipeline.

Most "paths" have different fields for some reason, which is why we have plenty of fields that are named very similar (like start and end timestamps), its based on the combination of test data, customer talks and the Crowdstrike event documentation (which lists all fields available in each path).

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml (hunk 5 above)
Comment (P1llus) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:52]: There was a very slight chance that there might only be one value, or something that might not split, however I could not proof either, so I added this as a failsafe, so that the field will stay keyword (even if its an array), and if it was not an array, I appended it instead (to the related.* fields)

Comment (efd6) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:52]: OK
